### ISSUE TITLE
error when no positional args

### ISSUE THREAD

**Issue by casperdcl** [iterative/shtab#29]
On `zsh` when there are no <positional> (only --optional) args, `shtab` doesn't work https://github.com/tldr-pages/tldr-python-client/pull/135#issuecomment-753408950

```zsh
% tldr <TAB>
_arguments:comparguments:312: doubled rest argument definition: *::args:->args
```

**Comment by bl-ue** [comment]
Any progress?

**Comment by casperdcl** [comment]
Ah sorry, will take a look in a couple of weeks (ping me again if I'm not responsive)

**Comment by bl-ue** [comment]
No problem (thanks for the lightnening fast response! ⚡⚡)

**Comment by bl-ue** [comment]
gentle ping, @casperdcl.  Any time so far? No problem if not 🙂

**Comment by casperdcl** [comment]
@bl-ue think it's fixed now :)

**Comment by yozachar** [comment]
Hi, this is the completion scripts that gets executed every time zsh is launched.

```zsh
#'tldr --print-completion zsh'
#compdef tldr

# AUTOMATICALLY GENERATED by `shtab`


_shtab_tldr_commands() {
  local _commands=(
    
  )
  _describe 'tldr commands' _commands
}

_shtab_tldr_options=(
  "(- : *)"{-h,--help}"[show this help message and exit]"
  "(- : *)"{-v,--version}"[show program\'s version number and exit]"
  "--search[Search for a specific command from a query]:search:"
  {-u,--update_cache}"[Update the local cache of pages and exit]"
  {-p,--platform}"[Override the operating system \[linux, osx, sunos, windows, common\]]:platform:(linux osx sunos windows common)"
  {-l,--list}"[List all available commands for operating system]"
  {-s,--source}"[Override the default page source]:source:"
  {-c,--color}"[Override color stripping]"
  {-r,--render}"[Render local markdown files]"
  {-L,--language}"[Override the default language]:language:"
  {-m,--markdown}"[Just print the plain page file.]"
  "(- : *)--print-completion[print shell completion script]:print_completion:(bash zsh tcsh)"
  "(*)::command to lookup:shtab_tldr_cmd_list"
)


_shtab_tldr() {
  local context state line curcontext="$curcontext" one_or_more='(-)*' remainder='(*)'

  if ((${_shtab_tldr_options[(I)${(q)one_or_more}*]} + ${_shtab_tldr_options[(I)${(q)remainder}*]} == 0)); then  # noqa: E501
    _shtab_tldr_options+=(': :_shtab_tldr_commands' '*::: :->tldr')
  fi
  _arguments -C -s $_shtab_tldr_options

  case $state in
    tldr)
      words=($line[1] "${words[@]}")
      (( CURRENT += 1 ))
      curcontext="${curcontext%:*:*}:_shtab_tldr-$line[1]:"
      case $line[1] in
        
      esac
  esac
}

# Custom Preamble
shtab_tldr_cmd_list(){
          _describe 'command' "($("/usr/bin/python" -m tldr --list | sed 's/[^[:alnum:]_]/ /g'))"
        }

# End Custom Preamble


typeset -A opt_args
_shtab_tldr "$@"
```

But I see this error in the console:

```console
_shtab_tldr:6: command not found: _arguments
```

**Comment by casperdcl** [comment]
Looks like a problem wirh your `zsh` install, vis. https://zsh-users.zsh.narkive.com/u1d54kbu/command-not-found-arguments
